sql/core/src/main/scala/org/apache/spark/sql/execution/command/cache.scala

```diff
@@ -17,30 +17,30 @@
 
 package org.apache.spark.sql.execution.command
 
-import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.sql.{AnalysisException, Dataset, Row, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
 import org.apache.spark.sql.catalyst.expressions.Attribute
 import org.apache.spark.sql.catalyst.plans.QueryPlan
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 
 case class CacheTableCommand(
-  tableName: String,
-  plan: Option[LogicalPlan],
-  isLazy: Boolean)
-  extends RunnableCommand {
+    tableIdent: TableIdentifier,
+    plan: Option[LogicalPlan],
+    isLazy: Boolean) extends RunnableCommand {
```

Comment:
Just added. : ) Please check if the checking is enough. https://github.com/apache/spark/pull/13572/files#diff-bc55b5f76add105ec32ae4107075b278R30

`default`.`tab` is still not allowed to create temp tables. Thus, I did not change that part. Let me know if anything else I need to change. Thanks!